qcodes/instrument/parameter.py

```diff
@@ -1745,15 +1730,174 @@ def setpoint_full_names(self) -> Optional[Sequence[Sequence[str]]]:
+        """
+        self._parameter.validate(value)
+        raw_value = self._parameter._from_value_to_raw_value(value)
+        self._update_with(value=value, raw_value=raw_value)
+
+    def _update_with(self, *,
+                     value: ParamDataType,
+                     raw_value: ParamRawDataType,
+                     timestamp: Optional[datetime] = None
+                     ) -> None:
+        """
+        Simply overwrites the value, raw value, and timestamp in this cache
+        with new ones.
+
+        Args:
+            value: new value of the parameter
+            raw_value: new raw value of the parameter
+            timestamp: new timestamp of the parameter; if ``None``,
+                then timestamp of "now" is used
+        """
+        self._value = value
+        self._raw_value = raw_value
+        if timestamp is None:
+            self._timestamp = datetime.now()
+        else:
+            self._timestamp = timestamp
+
+    def get(self, get_if_invalid: bool = True) -> ParamDataType:
+        """
+        Return cached value if time since get was less than ``max_val_age``,
+        otherwise perform ``get()`` on the parameter and return result. A
+        ``get()`` will also be performed if the parameter has never been
+        captured but only if ``get_if_invalid`` argument is ``True``.
+
+        Args:
+            get_if_invalid: if set to ``True``, ``get()`` on a parameter
+                will be performed in case the cached value is invalid (for
+                example, due to ``max_val_age``, or because the parameter has
+                never been captured)
+        """
+        no_get = not hasattr(self._parameter, 'get')
+
+        # the parameter has never been captured so `get` it but only
+        # if `get_if_invalid` is True
+        if self._timestamp is None:
+            if get_if_invalid:
+                if no_get:
+                    raise RuntimeError(f"Value of parameter "
+                                       f"{(self._parameter.full_name)} "
+                                       f"is unknown and the Parameter does "
+                                       f"not have a get command. "
+                                       f"Please set the value before "
+                                       f"attempting to get it.")
+                return self._parameter.get()
+            else:
+                return self._value
+
+        if self._max_val_age is None:
+            # Return last value since max_val_age is not specified
+            return self._value
+        else:
+            if no_get:
+                # TODO: this check should really be at the time of setting
+                #  max_val_age unfortunately this happens in init before
+                #  get wrapping is performed.
+                raise RuntimeError("`max_val_age` is not supported for a "
+                                   "parameter without get command.")
+
+            oldest_ok_val = (datetime.now()
```

Comment:
I know that you did not invent this, but I feel that the name of this variable is not good: `..._val` suggests that  it is a value. Maybe `last_accepted_ts` would be better.

Reply:
sure, will change :)